File: daily question/334. 递增的三元子序列/Increasing Triplet Subsequence.java
class Solution {
    /**
     * 贪心算法
     * @param nums
     * @return boolean
     */
    public boolean increasingTriplet(int[] nums) {
        int v1 = nums[0];
        int v2 = Integer.MAX_VALUE;
        for (int j = 1; j < nums.length; j++) {
            v1 = Math.min(nums[j], v1);
            if (nums[j] > v1) {
                v2 = Math.min(v2, nums[j]);
            }
            if (nums[j] > v2) {
                return true;
            }
        }
        return false;
    }
}

File: daily question/334. 递增的三元子序列/Increasing Triplet Subsequence.py

# TODO 解决测试用例超时问题
def increasingTriplet(nums: list[int]) -> bool:
    flag = False
    if len(nums) < 3:
        return flag

    for i in range(len(nums)-2):
        for j in range(i,len(nums)-1):
            for k in range(j,len(nums)):
                if i<j<k and nums[i]<nums[j]<nums[k]:
                    flag = True
                    break

    return flag
